<?xml version="1.0"?>
<launch>
    <!--File Path-->
    <arg name="image_path"     default="$(env HOME)/calibration_images/2021-07-27-13-49-10/"/>
    <arg name="save_directory" default="$(arg image_path)"/>

    <!--Number of corners-->
    <arg name="board_width"   default="8"/>
    <arg name="board_height"  default="5"/>
    <arg name="square_length" default="45.0"/><!--mm-->

    <arg name="image_width"  default="2448"/>
    <arg name="image_height" default="2048"/>

    <!--Calibration modes-->
    <!--0 Checker-->
    <!--1 Charuco-->
    <arg name="method" default="1"/>

    <!--Difference Threshold-->
    <arg name="threshold" default="0"/><!--px-->

    <!--Charuco marker length if required-->
    <arg name="marker_length" default="23.0"/><!--mm-->

    <!--Dictionary ID-->
    <arg name="dictionary" default="0"/><!--DICT_4X4_50-->

    <node name="stereo_calibration_node" pkg="stereo_calibration" type="stereo_calibration_node" output="screen">
        <param name ="image_path"      value="$(arg image_path)"/>
        <param name ="save_directory"  value="$(arg save_directory)"/>

        <param name ="board_width"  value="$(arg board_width)"/>
        <param name ="board_height" value="$(arg board_height)"/>
        <param name ="square_length" value="$(arg square_length)"/>

        <param name ="image_width"  value="$(arg image_width)"/>
        <param name ="image_height" value="$(arg image_height)"/>

        <param name ="method" value="$(arg method)"/>

        <param name ="threshold" value="$(arg threshold)"/>

        <param name ="marker_length" value="$(arg marker_length)"/>

        <param name ="dictionary" value="$(arg dictionary)"/>
    </node>
</launch>
